vm-setup/roles/v1aX_integration_test/tasks/node_reuse.yml

```diff
@@ -182,145 +212,209 @@
     ignore_errors: yes
 
   - name: Wait until controlplane is scaled down and "{{ NUMBER_OF_BMH_KCP }}" BMHs' are Ready.
-    shell: kubectl get bmh -n "{{ NAMESPACE }}" | grep -w ready | wc -l
-    environment:
-      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    k8s_info:
+      api_version: metal3.io/v1alpha1
+      kind: BareMetalHost
+      namespace: "{{ NAMESPACE }}"
+      kubeconfig: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    register: all_ready_hosts
     retries: 200
     delay: 20
-    register: all_ready_hosts
-    until: all_ready_hosts.stdout == "3"
+    until: all_ready_hosts.resources | filter_provisioning("ready") | length == 3
 
   - name: Scale worker up to "{{ NUMBER_OF_BMH_MD }}" to start testing MachineDeployment node reuse test scenario.
     shell: |
         kubectl scale machinedeployment "{{ CLUSTER_NAME }}" -n "{{ NAMESPACE }}" --replicas=1
     environment:
       KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    #- name: Scale worker up to "{{ NUMBER_OF_BMH_MD }}" to start testing MachineDeployment node reuse test scenario.
+    #  k8s_scale:
+    #    api_version: v1alpha3
+    #    kind: MachineDeployment
+    #    namespace: "{{ NAMESPACE }}"
+    #    name: "{{ CLUSTER_NAME }}"
+    #    kubeconfig: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    #    replicas: 1
 
   - name: Wait until "{{ NUMBER_OF_BMH_MD }}" more BMH is provisioned.
-    shell: kubectl get bmh -n "{{ NAMESPACE }}" | grep -w provisioned | wc -l
-    environment:
-      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    k8s_info:
+      api_version: metal3.io/v1alpha1
+      kind: BareMetalHost
+      namespace: "{{ NAMESPACE }}"
+      kubeconfig: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    register: provisioned_host
     retries: 200
     delay: 20
-    register: provisioned_host
-    until: provisioned_host.stdout == "2"
+    until: provisioned_host.resources | filter_provisioning("provisioned") | length == 2
```

Comment:
Checking this I have found that:
`NUMBER_OF_BMH_MD: "{{ NUM_OF_WORKER_REPLICAS|int }}"`
While:
`./scripts/feature_tests/node_reuse/Makefile:2:export NUM_OF_WORKER_REPLICAS := 1`

Which is not equal to the actual value `2` ! I think the var in the task name means 1 more BMH to be provisioned while another one already provisioned. @furkatgofurov7 probably can confirm and suggest if possible using a var instead of static value in these two tasks

Reply:
Oh interesting! Maybe we could set it to `NUM_OF_WORKER_REPLICAS + NUMBER_OF_BMH_MD`?

Reply:
```
---
  - set_fact:
      NUMBER_OF_BMH_KCP: "{{ NUM_OF_MASTER_REPLICAS|int }}"
      NUMBER_OF_BMH_MD: "{{ NUM_OF_WORKER_REPLICAS|int }}"
      NUMBER_OF_ALL_BMH: "{{ NUM_OF_WORKER_REPLICAS|int + NUM_OF_MASTER_REPLICAS|int }}"
```
`NUMBER_OF_BMH_MD` is initialized with `NUM_OF_WORKER_REPLICAS` It may work for this case but if vars values chaged I am afraid it won't